sw/device/tests/sim_dv/i2c_device_tx_rx_test.c

```diff
@@ -0,0 +1,225 @@
+ * to indicate the address that will be listened to by the device.
+ */
+static volatile const uint8_t kI2cDeviceAddress0 = 0x55;
+static volatile const uint8_t kI2cDeviceMask0 = 0x7f;
+static volatile const uint8_t kI2cDeviceAddress1 = 0x7f;  // disable match on
+                                                          // second address
+static volatile const uint8_t kI2cDeviceMask1 = 0x00;
+
+/**
+ * This symbol is meant to be backdoor loaded by the testbench.
+ * to indicate the number of bytes that should be sent.
+ *
+ * Because the test doesn't manage the FIFO during transaction, there's a limit
+ * to the number of bytes we can loopback in the test. I2C_PARAM_FIFO_DEPTH - 4
+ */
+static volatile const uint8_t kI2cByteCount = 0;
+
+static volatile bool tx_empty_irq_seen = false;
+static volatile bool trans_complete_irq_seen = false;
+
+/**
+ * Provides external irq handling for this test.
+ *
+ * This function overrides the default OTTF external ISR.
+ */
+void ottf_external_isr(void) {
+  plic_isr_ctx_t plic_ctx = {.rv_plic = &plic,
+                             .hart_id = kTopEarlgreyPlicTargetIbex0};
+
+  i2c_isr_ctx_t i2c_ctx = {
+      .i2c = &i2c,
+      .plic_i2c_start_irq_id = kTopEarlgreyPlicIrqIdI2c0FmtWatermark,
+      .expected_irq = 0,
+      .is_only_irq = false};
+
+  top_earlgrey_plic_peripheral_t peripheral;
+  dif_i2c_irq_t i2c_irq;
+  isr_testutils_i2c_isr(plic_ctx, i2c_ctx, &peripheral, &i2c_irq);
+
+  switch (i2c_irq) {
+    case kDifI2cIrqTxEmpty:
+      tx_empty_irq_seen = true;
+      i2c_irq = kDifI2cIrqTxEmpty;
+      break;
+    case kDifI2cIrqTransComplete:
+      trans_complete_irq_seen = true;
+      i2c_irq = kDifI2cIrqTransComplete;
+      break;
+    default:
+      LOG_ERROR("Unexpected interrupt (at I2C): %d", i2c_irq);
+      break;
+  }
+}
+
+void check_addr(uint8_t addr, dif_i2c_id_t id0, dif_i2c_id_t id1) {
+  CHECK(((addr & id0.mask) == id0.address) ||
+        ((addr & id1.mask) == id1.address));
+}
+
+bool test_main(void) {
+  if (kI2cByteCount > I2C_PARAM_FIFO_DEPTH - 4) {
+    LOG_ERROR(
+        "Test cannot fit %d bytes, 2 START records, and 2 STOP records in "
+        "buffers of depth %d",
+        kI2cByteCount, I2C_PARAM_FIFO_DEPTH);
+  }
+  CHECK_DIF_OK(
+      dif_i2c_init(mmio_region_from_addr(TOP_EARLGREY_I2C0_BASE_ADDR), &i2c));
+  CHECK_DIF_OK(dif_pinmux_init(
+      mmio_region_from_addr(TOP_EARLGREY_PINMUX_AON_BASE_ADDR), &pinmux));
+  CHECK_DIF_OK(dif_rv_plic_init(
+      mmio_region_from_addr(TOP_EARLGREY_RV_PLIC_BASE_ADDR), &plic));
+
+  CHECK_DIF_OK(dif_rv_plic_irq_set_enabled(
+      &plic, kTopEarlgreyPlicIrqIdI2c0TxEmpty, kTopEarlgreyPlicTargetIbex0,
+      kDifToggleEnabled));
+  CHECK_DIF_OK(dif_rv_plic_irq_set_enabled(
+      &plic, kTopEarlgreyPlicIrqIdI2c0TransComplete,
+      kTopEarlgreyPlicTargetIbex0, kDifToggleEnabled));
+  CHECK_DIF_OK(dif_rv_plic_irq_set_priority(
+      &plic, kTopEarlgreyPlicIrqIdI2c0TxEmpty, 0x1));
```

Comment:
nit: maybe use `kDifRvPlicMaxPriority` here?

Reply:
Thanks, I copied this from the i2c_host_tx_rx_test so I'll also fix it there.